### PR TITLE
Direct ILL absolute unit normalisation

### LINKED ISSUE #32384: Absolute units wrong in DirectILLReduction
**Describe the bug**

In `DirectILLReduction`,  [_absoluteUnits invokation](https://github.com/mantidproject/mantid/blob/34a591f251cc2edf65d83f802cb258a577bbb3ea/Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/DirectILLReduction.py#L500) doesn't match its [definition ](https://github.com/mantidproject/mantid/blob/34a591f251cc2edf65d83f802cb258a577bbb3ea/Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/DirectILLReduction.py#L24)

Using that option obviously doesn't work.

**To Reproduce**
Use the algorithm with AbsoluteUnits=On

### PR DESCRIPTION
**Pull request by gvardany**
**Description of work.**
Fixes the absolute unit normalisation code in DirectILLReduction algorithm.
The invocation was not corresponding to the signature.

**To test:**
Code review and `ctest -R DirectILLReduction`

Fixes #32384 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/DirectILLReduction.py [hunk 1]
```diff
@@ -28,9 +28,13 @@ def _absoluteUnits(ws, vanaWS, wsNames, wsCleanup, report, algorithmLogging):
     vanaMaterial = vanaWS.sample().getMaterial()
     vanaNumberDensity = vanaMaterial.numberDensity
     vanaCrossSection = vanaMaterial.totalScatterXSection()
+    if vanaNumberDensity == 0 or math.isnan(vanaNumberDensity) or math.isinf(vanaNumberDensity):
+        raise RuntimeError('Invalid vanadium number density, consider setting the material before: {}'.format(vanaNumberDensity))
+    if sampleNumberDensity == 0 or math.isnan(sampleNumberDensity) or math.isinf(sampleNumberDensity):
+        raise RuntimeError('Invalid sample number density, consider setting the sample material before: {}'.format(sampleNumberDensity))
+    if vanaCrossSection <= 0 or math.isnan(vanaCrossSection) or math.isinf(vanaCrossSection):
+        raise RuntimeError('Invalid vanadium cross-section, consider setting the material before: {}'.format(vanaCrossSection))
```
Comment (Harrietbrown) [inline Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/DirectILLReduction.py:36]: With these errors being changed could you put in some tests to check for invalid values for vanaNumberDensity, sampleNumberDensity and vanaCrossSection.

Comment (gvardany) [reply Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/DirectILLReduction.py:36]: Done.

**Comment by gvardany** [comment]
@Harrietbrown I added the tests you requested, could you have another look?

**Comment by gvardany** [comment]
done.

**Review by Harrietbrown — DISMISSED** [review]
Code looks good.
Good Job

**Comment by sf1919** [comment]
Please can you put the release note in a separate file.

**Comment by gvardany** [comment]
Done, can you reapprove?

**Review by Harrietbrown — APPROVED** [review]
Good Job

**Pull request merged**